packages/api/src/routes/middlewares/auth.ts

```diff
@@ -9,49 +11,77 @@ import { getCxIdOrFail } from "../util";
  * Process the API key and get the customer id.
  * The customer id is stored on the Request, property 'cxId'.
  */
-export const processAPIKey = (req: Request, res: Response, next: NextFunction): void => {
+export function processCxId(req: Request, res: Response, next: NextFunction): void {
+  const { log } = out("processCxId");
   try {
     // Just gets the cxId from the API Key, the actual auth is done on API GW.
     // Downstream routes should check whether `cxId` is present on the request or not.
     const encodedApiKey = req.header("x-api-key");
     req.cxId = getCxIdFromApiKey(encodedApiKey);
+    // TODO 1935 remove this
+    // TODO 1935 remove this
+    // TODO 1935 remove this
+    log(`..... cxId from API Key: ${req.cxId}`);
   } catch (error) {
-    // noop - auth is done on API GW level, this is just to make data available downstream
+    try {
+      // TODO 1935 remove this
+      // TODO 1935 remove this
+      // TODO 1935 remove this
+      req.cxId = getCxIdFromJwt(req);
+      log(`..... cxId from JWT: ${req.cxId} 🤘`);
+    } catch (error) {
+      // noop - auth is done on API GW level, this is just to make data available downstream
+    }
   }
   next();
-};
+}
 
 export function getCxIdFromApiKey(encodedApiKey: string | undefined): string {
   if (!encodedApiKey) throw new Error("No API Key provided");
   const apiKey = base64ToString(encodedApiKey);
   const splitApiKey = apiKey.split(":");
   if (splitApiKey.length !== 2) throw new Error("Invalid API Key format");
   const cxId = splitApiKey[1];
-  if (cxId.trim().length < 5) throw new Error("Invalid API Key format");
+  if (!isValidCxId(cxId)) throw new Error("Invalid API Key format");
+  return cxId;
+}
+
+export function getCxIdFromJwt(req: Request): string {
+  const jwtStr = req.header("Authorization");
+  if (!jwtStr) throw new Error("Missing token");
+  const rawToken = jwt.decode(jwtStr);
+  if (!rawToken) throw new Error("Invalid token");
+  const token = (typeof rawToken === "string" ? JSON.parse(rawToken) : rawToken) as jwt.JwtPayload;
+  const cxId = token["cognito:username"] ?? token["cxId"];
```

Comment:
why not `sub` vs `username`? I guess it doesn't matter as we're deprecating this anyway yea?